### PR TITLE
Follow-ups to the switch to Lerna

### PR DESCRIPTION
**Pull request by edmorley**
* Removes the now unused publish script and its dependencies.
* Removes the `lerna bootstrap` from the Travis run, since it's equivalent to the `yarn install` already run prior.
* Removes `yarn.lock` from the package's `.npmignore`, since there is now only the repo root instance of the file.
* Updates the development docs to reflect the new workflow.
* Sets node and yarn version ranges in `package.json` to help avoid issues caused by using a version not recommended by the docs.
* Adds `yarn release` and `release:preview` aliases using lerna, for easier publishing.
* Removes version and unnecessary `devDependencies` from root `package.json`.

### CODE REVIEW

**Comment by edmorley** [comment]
I've also cleared the Travis caches for this repo, to reduce the bloat from old package versions, so once re-populated after the next master push, between that and the switch to lerna, the travis run setup phase should be pretty fast :-)

File: package.json [hunk 1]
```diff
@@ -7,6 +7,10 @@
   "homepage": "https://neutrino.js.org",
   "bugs": "https://github.com/mozilla-neutrino/neutrino-dev/issues",
   "repository": "mozilla-neutrino/neutrino-dev",
+  "engines": {
+    "node": ">=6.1.0",
```
Comment (eliperelman) [inline package.json]: Should this be `>=6.10.0`?

Comment (edmorley) [reply package.json]: Ah yes it should, good spot :-)

**Comment by edmorley** [comment]
I've updated the PR to address that review comment, but have also added a few other things (see the OP for what's new), which made it a bit longer, so I've now split up into separate commits for easer review.

File: package.json [hunk 2]
```diff
@@ -21,18 +24,14 @@
...
     "docs:serve": "gitbook serve",
     "precommit": "lint-staged",
-    "release": "node .scripts/publish",
+    "release": "lerna publish --force-publish=*",
+    "release:preview": "lerna publish --force-publish=* --skip-git --skip-npm",
     "test": "nyc --reporter lcov ava packages/*/test"
   },
   "devDependencies": {
     "ava": "^0.22.0",
-    "babel-eslint": "^8.0.1",
```
Comment (eliperelman) [inline package.json]: Removing all these deps is nicer now that hoisting is in place.

**Pull request merged**